paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerCompletesAdvancementScriptEventPaperImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.player.PlayerCompletesAdvancementScriptEvent;
+import com.denizenscript.denizen.paper.PaperModule;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.ListTag;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import net.kyori.adventure.text.Component;
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.player.PlayerAdvancementDoneEvent;
+
+public class PlayerCompletesAdvancementScriptEventPaperImpl extends PlayerCompletesAdvancementScriptEvent {
+
+    public PlayerCompletesAdvancementScriptEventPaperImpl() {
+        instance = this;
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        if (name.equals("criteria")) {
+            ListTag criteria = new ListTag();
+            criteria.addAll(event.getAdvancement().getCriteria());
+            return criteria;
+        }
+        else if (name.equals("advancement")) {
+            return new ElementTag(event.getAdvancement().getKey().getKey());
+        }
+        else if (name.equals("message")) {
+            return new ElementTag(PaperModule.stringifyComponent(event.message(), ChatColor.WHITE));
+        }
+        return super.getContext(name);
+    }
+
+    @Override
+    public boolean applyDetermination(ScriptPath path, ObjectTag determinationObj) {
+        String determination = determinationObj.toString();
+        String lower = CoreUtilities.toLowerCase(determination);
+        if (lower.startsWith("no_message")) {
+            event.message(null);
+            return true;
+        }
+        else if (event instanceof PlayerAdvancementDoneEvent) {

```

Comment:
That check is redundant, `event` is always an instanceof `PlayerAdvancementDoneEvent`